jni/include/FaissIndexBQ.h

```diff
@@ -0,0 +1,177 @@
+#include "faiss/utils/hamming_distance/hamdis-inl.h"
+#include <vector>
+#include <iostream>
+#include <cassert>
+
+namespace knn_jni {
+    namespace faiss_wrapper {
+        struct ADCFlatCodesDistanceComputer1Bit : faiss::FlatCodesDistanceComputer {
+            static constexpr int BATCH_SIZE = 8;
+            static constexpr int NUM_POSSIBILITIES_PER_BATCH = 1 << BATCH_SIZE;
+            const uint8_t* codes;
+            const float* query;
+            int dimension;
+            size_t code_size;
+            faiss::MetricType metric_type;
+            std::vector<std::vector<float>> lookup_table;
+            std::vector<float> coord_scores; // scores for each dimension
+            float correction_amount; // used in batched distances
+
+            ADCFlatCodesDistanceComputer1Bit(const uint8_t* codes, size_t code_size, int d,
+                faiss::MetricType metric_type = faiss::METRIC_L2) 
+                : FlatCodesDistanceComputer(codes, code_size), dimension(d), query(nullptr), metric_type(metric_type) {
+                this->codes = codes;
+                this->code_size = code_size;
+                this->dimension = d;
+                correction_amount = 0.0f;
+            }
+
+            float distance_to_code_l2_unbatched(const uint8_t* code) {
+                // L2 distance
+               float score = 0.0f;
+               for (int i = 0; i < dimension; i++) {
+                   uint8_t code_block = code[(i / 8)];
+                   int bit_offset = 7 - (i % 8);
+                   int bit_mask = 1 << bit_offset;
+                   int code_masked = (code_block & bit_mask);
+                   int code_translated = code_masked >> bit_offset;
+
+                   float dim_score = (code_translated - query[i]) * (code_translated - query[i]);
+
+                   score += dim_score;
+               }
+               return score;       
+               }
+
+            float distance_to_code_batched(const uint8_t * code) {
+                float dist = 0.0f;
+                for (int i = 0 ; i < dimension / 8; ++i) {
+                    const unsigned char code_batch = code[i];
+                    dist += this->lookup_table[i][code_batch];
+                }
+
+                return dist + correction_amount; 
+            }
+
+            virtual float distance_to_code(const uint8_t* code) override {
+                return distance_to_code_batched(code);
+            };
+
+            void compute_cord_scores() {
+                // make sure we've already set the query
+                assert(this->query != nullptr);
+                this->coord_scores = std::vector<float>(this->dimension, 0.0f);
+                if (this->metric_type == faiss::METRIC_L2) {
+                    compute_cord_scores_l2();
+                } else if (this->metric_type == faiss::METRIC_INNER_PRODUCT) {
+                    compute_cord_scores_inner_product();
+                }
+                
+                else {
```

Comment:
fix syntax

Reply:
Do you mean remove the redundant else statement? Or is there something else wrong with syntax?